src/brsTypes/components/RoSGNode.ts

```diff
@@ -533,6 +545,31 @@ export class RoSGNode extends BrsComponent implements BrsValue, BrsIterable {
         return false;
     }
 
+    /* used for isSubtype */
+    protected setExtendsType() {
+        let baseClass = this.constructor;
+        let currentNodeType: string, parentType: string;
+        while (baseClass) {
+            currentNodeType = baseClass.name.toLowerCase();
+
+            const parentClass = Object.getPrototypeOf(baseClass);
```

Comment:
I don't think the method prototypes is the best way to do this since we might for whatever reason wrap these objects. The _createNodeByType_ function has context on the node's inheritance chain, and passes its subtype into the constructor. I suggest also passing in the superclass name as well, so you don't need to do this lookup.

Reply:
I'd say you could also use `interpreter.environment.nodeDefMap` here for non-builtin components. If it's builtin (i.e. `Group`), then the simplest is probably `Object.getPrototypeOf`. If it's custom (i.e. `FooComponent`), then the simplest is probably `interpreter.environment.nodeDefMap.get(subtype).extends`.

[We do a similar recursion in the component processor.](https://github.com/sjbarag/brs/blob/main/src/componentprocessor/index.ts#L123)

Reply:
@lkipke I'll update to use the nodeDefMap when needed.

Reply:
I elected to update the subTypeHierarchy on build/creation, as it should never change based on the current state of the interpreter